process_report/invoices/pi_specific_invoice.py

```diff
@@ -0,0 +1,38 @@
+import os
+from dataclasses import dataclass
+
+import pandas
+
+import process_report.invoices.invoice as invoice
+import process_report.util as util
+
+
+@dataclass
+class PIInvoice(invoice.Invoice):
+    def export(self):
+        self.local_to_s3_mapping = list()
+        if not os.path.exists(
+            self.name
+        ):  # self.name is name of folder storing invoices
+            os.mkdir(self.name)
+
+        pi_list = self.data[invoice.PI_FIELD].unique()
+
+        for pi in pi_list:
+            if pandas.isna(pi):
+                continue
+            pi_projects = self.data[self.data[invoice.PI_FIELD] == pi]
```

Comment:
This seems like a great candidate to be split into a separate method that takes a pi as an argument and exports that PI only.

Reply:
@QuanMPhm how goes the splitting of this section?

Reply:
It was resolved last week.